crates/tako/src/server/core.rs

```diff
@@ -21,6 +22,7 @@ pub type CustomConnectionHandler = Box<dyn Fn(ConnectionDescriptor)>;
 pub struct Core {
     tasks: Map<TaskId, TaskRef>,
     workers: Map<WorkerId, Worker>,
+    event_queue: Vec<MonitoringEvent>,
```

Comment:
Let's create some struct in `monitoring.rs` that will handle storing and fetching monitoring events. It shouldn't be the responsibility of `Core` nor the client code. `Core` will just hold this struct inside of it.